edgelet/edgelet-docker/src/notary.rs

```diff
@@ -1,90 +1,132 @@
-use std::path::{Path,PathBuf};
-use log::{info, debug};
+use crate::{Error, ErrorKind};
+use failure::ResultExt;
+use log::{debug, info};
 use serde_json::json;
 use std::fs;
 use std::fs::File;
-use crate::{Error,ErrorKind};
-use failure::ResultExt;
+use std::path::{Path, PathBuf};
 
-pub fn notary_init(homedir : &Path, hostname : String, path : PathBuf) -> Result<(), Error> {
-  // Validate inputs
-  if hostname.is_empty() {
-    return Err(ErrorKind::InitializeNotary("hostname is empty".to_owned()).into());
-  }
+pub fn notary_init(homedir: &Path, hostname: &str, path: &PathBuf) -> Result<PathBuf, Error> {
```

Comment:
home_dir, host_name for the naming. 

Also the "host_name" is not very meaningful in this context. I think you meant the registry url, right?

Reply:
I have renamed it as registry_server_hostname for more clarification